server/src/test/java/com/octopus/teamcity/opentelemetry/server/LogMaskerTest.java

```diff
@@ -2,41 +2,91 @@
 
 import org.apache.commons.lang3.RandomStringUtils;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 
 import java.util.Locale;
-import java.util.Random;
 
 import static org.junit.jupiter.api.Assertions.*;
 
 class LogMaskerTest {
 
     @Test
-    void testMaskLowerCase() {
+    void LowerCaseApiKeyGetsMasked() {
         int APIKEY_LENGTH = 31;
         String generatedString = RandomStringUtils.randomAlphanumeric(APIKEY_LENGTH).toLowerCase(Locale.ROOT);
+        assertFalse(LogMasker.mask(generatedString).contains(generatedString));
         assertEquals(LogMasker.API_KEY_REPLACEMENT_REGEX, LogMasker.mask(generatedString));
     }
 
     @Test
-    void testMaskUpperCase() {
+    void UpperCaseApiKeyNotMasked() {
         int APIKEY_LENGTH = 31;
         String generatedString = RandomStringUtils.randomAlphanumeric(APIKEY_LENGTH).toUpperCase(Locale.ROOT);
+        assertTrue(LogMasker.mask(generatedString).contains(generatedString));
         assertNotEquals(LogMasker.API_KEY_REPLACEMENT_REGEX, LogMasker.mask(generatedString));
     }
 
-    @Test
-    void testMaskLengthShort() {
-        Random random = new Random();
-        int API_KEY_LENGTH = random.nextInt(30);
-        String generatedString = RandomStringUtils.randomAlphanumeric(API_KEY_LENGTH).toUpperCase(Locale.ROOT);
+    @ParameterizedTest
+    @ValueSource(ints = { 31, 40, 70, 90, 100 })
+    void ApiKeyMaskedForLowerCaseStringLengthGreaterThan31(int apiKeyLength) {
+        String generatedString = RandomStringUtils.randomAlphanumeric(apiKeyLength).toLowerCase(Locale.ROOT);
+        assertFalse(LogMasker.mask(generatedString).contains(generatedString));
+        assertTrue(LogMasker.mask(generatedString).contains(LogMasker.API_KEY_REPLACEMENT_REGEX));
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = { 1, 10, 20, 25, 30 })
+    void ApiKeyNotMaskedForLowerCaseStringLengthLessThan31(int apiKeyLength) {
+        String generatedString = RandomStringUtils.randomAlphanumeric(apiKeyLength).toLowerCase(Locale.ROOT);
+        assertTrue(LogMasker.mask(generatedString).contains(generatedString));
+        assertFalse(LogMasker.mask(generatedString).contains(LogMasker.API_KEY_REPLACEMENT_REGEX));
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = { 10, 20, 70, 90, 150 })
```

Comment:
```suggestion
    @ValueSource(ints = { 30, 31, 32 })
```
Consider deleting the other test which just asserts on `31`

Reply:
updated to this thanks!